### PR TITLE
Order CSSFontSelector::clients_ and add assertions

### PR DESCRIPTION
**Pull request by kannanvijayan**
For RUN-1728 (https://linear.app/replay/issue/RUN-1728/order-cssfontselectorclients-and-add-more-assertions-around-it)

Bucket issue RUN-1219

### CODE REVIEW

File: third_party/blink/renderer/core/css/css_font_selector.cc [hunk 1]
```diff
@@ -68,12 +69,20 @@ UseCounter* CSSFontSelector::GetUseCounter() const {
 void CSSFontSelector::RegisterForInvalidationCallbacks(
     FontSelectorClient* client) {
   CHECK(client);
+  recordreplay::Assert("[RUN-1219-1728] CSSFontSelector::RegisterForInvalidationCallbacks");
   clients_.insert(client);
+  if (recordreplay::IsRecordingOrReplaying("RUN-1219-CSSFontSelector")) {
```
Comment (Domiii) [inline third_party/blink/renderer/core/css/css_font_selector.cc:74]: This requires a change to `v8`'s `api.cc`: `gKnownFeatures` (or what it's called?)
I'd recommend using the general-purpose label that already exists.
→ [Reference PR](https://github.com/replayio/chromium/pull/371/files) for making these types of changes.

File: third_party/blink/renderer/core/css/css_font_selector.cc [hunk 2]
```diff
@@ -176,6 +185,9 @@ bool CSSFontSelector::IsAlive() const {
 void CSSFontSelector::Trace(Visitor* visitor) const {
   visitor->Trace(tree_scope_);
   visitor->Trace(clients_);
+  if (recordreplay::IsRecordingOrReplaying("RUN-1219-CSSFontSelector")) {
```
Comment (Domiii) [inline third_party/blink/renderer/core/css/css_font_selector.cc:188]: This `if` is not necessary

**Review by sgeisenh — APPROVED** [review]

File: third_party/blink/renderer/core/css/css_font_selector.cc (hunk 1 above)
Comment (sgeisenh) [reply third_party/blink/renderer/core/css/css_font_selector.cc:74]: Was `"avoid-weak-pointers"` already added before that PR?

Comment (kannanvijayan) [reply third_party/blink/renderer/core/css/css_font_selector.cc:74]: Yeah it's there on v8's tip so I'll just use that.

**Pull request merged**